src/org/labkey/test/pages/DatasetInsertPage.java

```diff
@@ -81,12 +83,17 @@ public void insert(Map<String, String> values, boolean expectSuccess, String err
 
         clickAndWait(Locator.lkButton("Submit"));
 
-        if (errorMsg != null && !errorMsg.isEmpty())
+        if (expectSuccess)
         {
-            if (!expectSuccess)
-                assertTextPresent(errorMsg);
-            else
-                assertTextNotPresent(errorMsg);
+            assertElementNotPresent(Locators.labkeyError);
+        }
+        else if (!StringUtils.isBlank(errorMsg))
+        {
+            assertTextNotPresent(errorMsg);
+        }
+        else
+        {
+            assertElementPresent(Locators.labkeyError);
         }
```

Comment:
This change should make that error check more general. Previously, it would only fail if that exact specified error appeared; now it will fail with any error.